### ISSUE TITLE
URL structure is very limiting

### ISSUE THREAD

**Issue by freaksauce** [tomoakley/freemeals.uk#87]
If you were to change the url structure to include the town/city users could send on a link. Using `/provider/1` will never allow sharing of links because it relies on the data being filtered first.

eg. `/bath` to then filter and return the vendor 'Bombay Balti'

**Comment by adlradlr** [comment]
It would be a nice QOL upgrade. However, a few key tasks would have to be done to get this working in the apps current state.

- Create a location to URL slug encoder that sanitized locations.
- Create a URL slug to location decoder.
- Check for the presence of URL params and check if the param is a valid location, if so set AppContext's selectedLocation as that location.

I think the difficulty comes from ensuring it's valid for every location. Consider the two locations "Barrow-in-Furness" and "Barton upon humber" that currently exist in the app's data. Adding hyphens in place of spaces in a location to get a URL, would mean that a decoder that adds in spaces in place of hyphens would invalidate some locations ex. "Barrow in Furness".

**Comment by freaksauce** [comment]
Yes I agree it's not a small task, the easiest part would probably be stripping the hyphens and spaces from all city/town values in both the url and api data in order to do a comparison check

**Comment by tomoakley** [comment]
yeah we need to do this. I was thinking of using the coordinates in the path, e.g `/<lat>,<lng>/provider/name` but your method is maybe better.
